plotting/plot_factory/plot_factory.py

```diff
@@ -0,0 +1,208 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Constructs a plot and DashApp object for insertion into directly into a web page
+"""
+
+# Note pandas and plotly.graph_objs dependencies are used, but not recognised by pycharm interpreter
+
+# Visualisation Dependencies
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objects as go  # pylint: disable=unused-import
+
+# Internal Dependencies
+from plotting.plot_meta_language.interpreter import Interpreter
+
+
+# Data Dependencies
+
+
+class PlotFactory:
+    """Creates figures from formatted data and layout
+       producing a DashApp for direct insertion inside a given web page
+       =_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=
+        """
+
+    @staticmethod
+    def get_trace_list(data, layout, figure_name):
+        """Creates trace list containing traces for each spectrum to place in figure
+
+        :param data
+        :param layout
+        :param figure_name
+
+        :return object
+
+
+        """
+        trace_list = []
+        for spectrum in data.index.unique():
+
+            trace_list.append(Trace(mode=layout.mode,
+                                    plot_style=layout.plot_type,
+                                    plot_name=f"{figure_name}_{layout.plot_type}",
+                                    data=data.loc[spectrum],
+                                    error_bars=layout.error_bars).trace)
+
+        return trace_list
+
+    def construct_plot(self, plot_meta_file_location, data, figure_name):
+        """Gets DashaApp after calling layout and trace to construct figure in figure factory
+
+           :param plot_meta_file_location (string)
+           :param data (pandas dataframe)
+           :param figure_name (string)
+
+           :return DashApp (object)
+           """
+
+        data.set_index('Spectrum', inplace=True)
+        layout = Layout(plot_meta_file_location)
+        trace_list = self.get_trace_list(data=data, layout=layout, figure_name=figure_name)
+        figure = dict(data=trace_list, layout=layout.layout)
+
+        return DashApp(figure=figure, app_id=figure_name)
+
+
+class Layout:
+    """ Extract Layout as dictionary from interpreted meta data """
+    def __init__(self, plot_style):
+        """
+        :param plot_style (dictionary)
+        """
+        self.meta_data = plot_style
+        self.mode = None
+        self.plot_type = None
+        self.error_bars = None
+        self.layout = self.extract_layout()
+
+    def read_plot_meta_data(self):
+        """Use plot interpreter to interpret plot meta data
```

Comment:
for docstrings can we follow the below structure:
```
"""
Description
:param ...:
:return: ...
"""
```
Note the indentation and no newlines